Framework/CurveFitting/src/Algorithms/NormaliseByPeakArea.cpp

```diff
@@ -283,29 +275,31 @@ double NormaliseByPeakArea::fitToMassPeak(const MatrixWorkspace_sptr &yspace,
 }
 
 /**
- * Divides the input Y & E data by the given factor
- * @param area Value to use as normalisation factor
- * @param index Index on input spectrum to normalise
- */
+* Divides the input Y & E data by the given factor
+* @param area Value to use as normalisation factor
+* @param index Index on input spectrum to normalise
+*/
 void NormaliseByPeakArea::normaliseTOFData(const double area,
                                            const size_t index) {
-  const auto &inY = m_inputWS->readY(index);
-  auto &outY = m_normalisedWS->dataY(index);
+  const auto &inY = m_inputWS->y(index);
+  std::vector<double> outY(m_normalisedWS->y(index).size());
   std::transform(inY.begin(), inY.end(), outY.begin(),
                  std::bind2nd(std::divides<double>(), area));
+  m_normalisedWS->mutableY(index) = outY;
```

Comment:
Why are you creating a temporary vector?

You can either do the transform on `HistogramY`, or use `operator/=`:

```cpp
m_normalisedWS->mutableY(index) /= area;
```

The latter might be slightly slower than doing the transform directly.